packages/extension-ui/src/Popup/Accounts/index.tsx

```diff
@@ -42,6 +43,48 @@ function Accounts({ className }: Props): React.ReactElement {
     setFilter(filter.toLowerCase());
   }, []);
 
+  const accountsByGenesisHash: { [key: string]: AccountWithChildren[] } = filteredAccount.reduce(
+    (result: { [key: string]: AccountWithChildren[] }, account) => {
+      const { genesisHash } = account;
+      const foundKey = Object.keys(knownGenesis).find((key) => knownGenesis[key].includes(genesisHash ?? ''));
+
+      if (!foundKey) {
+        if (!result.any) {
+          result.any = [];
+        }
+
+        result.any.push(account);
+
+        return result;
+      }
+
+      if (!result[foundKey]) {
+        result[foundKey] = [];
+      }
+
+      const accountWithChildren: AccountWithChildren = { ...account };
+
+      if (account.parentAddress) {
+        const parentAccount = filteredAccount.find(({ address }) => address === account.parentAddress);
+
+        if (parentAccount) {
+          if (!parentAccount.children) {
+            parentAccount.children = [];
+          }
+
+          parentAccount.children.push(accountWithChildren);
```

Comment:
I don't really get that - you should not make any side effects in reduce - it should be always modification of the current value and return accumulator. Here it seems you are not modifying currently parsed result but are changing something in another place that's not related to the value returned by the result. This can have unexpected consequences and should be avoided.